### PR TITLE
BUGFIX: Prevent Browser bugs from affecting number inputs

### LINKED ISSUE #297: input type="number" erase bug in Chrome
### Environment

* Elixir version (elixir -v): 1.9.1
* Phoenix version (mix deps): 1.4.8
* NodeJS version (node -v): 12.7.0
* NPM version (npm -v): 6.10.0 
* Operating system: Ubuntu 19.04
* Browser: Chromium: Version 76.0.3809.87 (64-bit)

### Actual behavior

I've created a [fork of the phoenix_live_view_example repo](https://github.com/LightningK0ala/phoenix_live_view_example) to show this problem. You can clone the repo, follow the setup instructions to run it and go to [http://localhost:4000/users/new](http://localhost:4000/users/new) on a Chrome browser.

1. Type "10.00" in the "Test" input field.
2. Erase the last 2 decimal digits.

As soon as you reach the dot, the cursor will jump to the beginning and the value will change to "10".

ezgif com-video-to-gif

This issue only happens on Chrome, I've tried it both on a Mac and Ubuntu.
Firefox doesn't have this issue.
Note I don't have any browser plugins installed on Chrome.

### Expected behavior

When erasing and reaching the dot, the cursor should not jump and the value should remain "10."

### PR DESCRIPTION
**Pull request by snewcomer**
### Fix

This PR takes the approach of updating the `input[number]` element if:
1. `badInput` is false
2. Not currently active element

By following these rules, we are letting the user continue working in the input without the browser interfering.

Chrome and Firefox treat number inputs differently.  In the past, Chrome even reformatted the separator.  This shows the historical context browser vendors took towards `input[number]`.  Today, many issues still exist.  Chrome/Firefox reformats inputs with `.` depending on the user input after we programatically update the input.  These are issues communities like React have faced and (still/where?) solved in a similar manner.  

### Alternate options?
We could look into preventing the "change" event from bubbling on the input.  But that might trap and not allow a user's listeners from firing.

Ref #300 #291 

close #297

### CODE REVIEW

**Comment by LightningK0ala** [comment]
Tested ACK

**Comment by leishman** [comment]
Awesome. This fixes our bug!

File: assets/js/phoenix_live_view.js [hunk 1]
```diff
@@ -704,6 +705,22 @@ let DOM = {
...
+          // report bad inputs so we want to do nothing
+          // eg. 1..
+          if (fromEl.validity && fromEl.validity.badInput) {
+            return false
+          }
+
+          // Moreover, Firefox will delete the separator when going from `1.1` to `1.`
+          // So we should never update the input if it is the currently active element
+          // This is the "safe" route and approaches other communities like React have taken
+          if (fromEl.ownerDocument.activeElement === fromEl) {
```
Comment (chrismccord) [inline assets/js/phoenix_live_view.js]: can we use the same `focused` check that we use elsewhere and that is in scope here?

**Comment by chrismccord** [comment]
@snewcomer I think this is the best approach for us to go with for now. Can you add docs to the main `PhoenixLiveView` moduledoc for our number handling? IE "invalid" numbers will be sent as blank input from the client. Also double check the existing `focused` variable we have in scope, then this is good to go. Thanks!

**Comment by chrismccord** [comment]
❤️❤️❤️🐥🔥

**Pull request merged**